coral/src/app/features/topics/details/overview/TopicOverview.tsx

```diff
@@ -25,12 +34,23 @@ function TopicOverview() {
   } = useTopicDetails();
 
   const {
-    topicInfo: { topicOwner = false, hasOpenTopicRequest },
+    topicInfo: { topicOwner = false, hasOpenTopicRequest, clusterId },
```

Comment:
We could add is as separate property to topic details? Also I feel this already has a lot of properties :D and it feels redundant to pass `topicInfo` and then some additional properties of `topicInfo`.

WDYT of rather making a follow up ticket and have a proper `useTopicDetails` hook instead of using the `Context`? The context is not really as type safe as we usually like I feel.

Reply:
>Something similar to useExtendedEnvironments

Yep! Using the outlet was very convenient in the beginning, but now we're passing a lot of stuff. I've not (yet?) found out how to type it 100% properly, so if I e.g. doing this (check out the last props on both sides) will be ok for TypeScript  (good thing we type our tests, because we would notice thanks to them)

<img width="879" alt="Screenshot 2023-07-13 at 16 42 29" src="https://github.com/aiven/klaw/assets/943800/d917ac60-cfc0-4b1c-b352-f8e7ab188ed2">

So maybe doing our own hook would be better?

Reply:
There is a world where this PR was merged and not closed, which would allow to use a generic to type `Outlet` https://github.com/remix-run/react-router/pull/8805

The only option now is probably to have a single shared type and cast with `as` in the `context` prop 

https://github.com/aiven/klaw/assets/20607294/0414e854-73f5-4af8-a3aa-7aa890cfcbe5

Reply:
I mean it would be not _really_ typesafe but more safe then now? 😬  so I would say we go for it?

Should I pass the connectorId as a separate property (string) for this PR or do we merge it as it is and replace the outlet hook with a custom hook in a later PR?

Reply:
I mean, what kind of safety would you expect? ^^ I don;'t think the `Outlet` context can ever infer the type of the `context` prop without someone telling it explicitly. It would be more idiomatic to do it as a generic, but from what I read, even doing it as a generic would only be a mask for the lib to do a `as` cast in the background :o

> Should I pass the connectorId as a separate property (string) for this PR or do we merge it as it is and replace the outlet hook with a custom hook in a later PR?

I think you can leave the stringifying as it is, it's really a minor issue ^^

Reply:
OK! Would you merge it?  (want to do a follow up PR to this :D)

![tenor-39604154](https://github.com/aiven/klaw/assets/943800/364d7061-67b3-4276-8966-21a44f735ddb)